com.unity.multiplayer.mlapi/Runtime/Connection/PendingClient.cs

```diff
@@ -8,22 +8,11 @@ public class PendingClient
         /// <summary>
         /// The ClientId of the client
         /// </summary>
-        public ulong ClientId;
-
+        public ulong ClientId { get; internal set; }
```

Comment:
do you think this API-level change is needed? should we guard the setter against public access? if so, is there a particular reason to do so?

Reply:
Yes, it should not be set externally. It's an oversight that it was not originally that way.

Reply:
I guess what I'm getting at is that just let's not try to kill two birds with one stone right.
introducing a new state and its implementation below (as it sounds) necessary to fix this issue but making changes to the public API is not.

we could do other things separately (in that second topic, I'd argue making PendingClient a struct instead of class and leaving both fields as fields instead of get/set properties etc. — but I don't want to argue or align on that here in this PR, that's what I'm getting at).